### PR TITLE
Do not retry to open cache store on ENOMEM

### PR DESCRIPTION
**Pull request by JosiahWI**
This replaces the `SocketManager::open` function with a new `safe_open` function in tscore that retries the call if it is interrupted a signal, but does not retry it if the kernel runs out of memory or buffers for a FIFO cannot be allocated. The change is split into three commits with descriptive comments.

### CODE REVIEW

**Comment by JosiahWI** [comment]
[approve ci freebsd]

File: src/tscore/ink_file.cc [hunk 1]
```diff
@@ -42,6 +42,21 @@ using ioctl_arg_t = union {
...
+ * This is a wrapper for the POSIX open() call that will retry the call if a
+ * transient error occurs.
+ */
+int
+safe_open(char const *path, int oflag, mode_t mode)
+{
+  int r{-1};
+  do {
+    r = open(path, oflag, mode);
+  } while ((r < 0) && (errno == EINTR));
```
Comment (JosiahWI) [inline src/tscore/ink_file.cc:56]: The `ENOBUFS` error code isn't checked because it's not documented as a possible error condition for `open()`.

**Comment by JosiahWI** [comment]
[approve ci freebsd]

File: include/tscore/ink_file.h [hunk 2]
```diff
@@ -80,6 +80,13 @@
 //
 #define INK_FILEPATH_TRUENAME 0x20
 
+namespace
```
Comment (ywkaras) [inline include/tscore/ink_file.h:83]: Combining an anonymous namespace with `inline` is redundant.

File: src/tscore/ink_file.cc [hunk 3]
```diff
@@ -42,6 +42,21 @@ using ioctl_arg_t = union {
...
+  return r;
+}
+
```
Comment (ywkaras) [inline src/tscore/ink_file.cc:59]: I'm not sure why this was originally inlined in the header.  But remember that functions declared with inline are not necessarily inlined.  If the compiler considers it non-optimal to actually inline, if will put the function in a vague linkage section, which prevents multiple copies of it from being present in the linked executable:  https://gcc.gnu.org/onlinedocs/gcc/Vague-Linkage.html .

Comment (JosiahWI) [reply src/tscore/ink_file.cc:59]: I think we wouldn't usually put definitions in the header file unless we think they may be useful to inline. Should we keep this in the header file with the `inline` declaration, then?

Comment (ywkaras) [reply src/tscore/ink_file.cc:59]: Personally I'm fine with it not being inline.

**Comment by ywkaras** [comment]
Looks good to me, just needs a non-Yahoo approval.

**Review by maskit — APPROVED** [review]

**Pull request merged**